src/rvsdg/egglog_optimizer/subst.rs

```diff
@@ -151,8 +151,196 @@ fn subst_beneath_rules() -> Vec<String> {
     res
 }
 
-pub(crate) fn subst_rules() -> String {
-    let mut res = vec![include_str!("subst.egg").to_string()];
+// Below, TYPE is one of {Expr, Operand, Body, VecOperand, VecVecOperand}
+
+// Generate rules to replace args via some procedure. See below for examples.
+//
+// Will generate functions named func_name_format.replace("{}", TYPE),
+// with parameter types [TYPE] ++ aux_param_types
+//
+// arg_rules are hardcoded rules for the Arg case of the function for Operand
+fn functions_modifying_args(
+    func_name_fmt: &str,
```

Comment:
We can't just have the user provide that string, since we also need to know how many args there are

Reply:
I see, okay
Can we at least move them so they are next to each other sequentially? They are both a part of the schema for the function

Reply:
Done!

(I kept it as `aux_param_types` rather than `function_types` to prefer verbosity / correctness, to make it clear it's not referring to the return type, and does not include all param types)